library/src/window_impl.cxx

```diff
@@ -285,11 +285,28 @@ void window_impl::UpdateDynamicOptions()
     this->Internals->Options.getAsBool("render.background.blur"));
   this->Internals->Renderer->SetBlurCircleOfConfusionRadius(
     this->Internals->Options.getAsDouble("render.background.blur.coc"));
-  this->Internals->Renderer->SetHDRIFile(
-    this->Internals->Options.getAsString("render.background.hdri"));
   this->Internals->Renderer->SetLightIntensity(
     this->Internals->Options.getAsDouble("render.light.intensity"));
 
+  std::string hdriFile = this->Internals->Options.getAsString("render.hdri.file");
+  bool hdriAmbient = this->Internals->Options.getAsBool("render.hdri.ambient");
+  bool hdriSkybox = this->Internals->Options.getAsBool("render.background.skybox");
+#ifndef F3D_NO_DEPRECATED
+  std::string legacyHDRI = this->Internals->Options.getAsString("render.background.hdri");
+  if (!legacyHDRI.empty())
+  {
+    hdriFile = legacyHDRI;
+    hdriAmbient = true;
+    hdriSkybox = true;
+    // TODO This is a bit spammy, how could we deprecate libf3d options cleanly ?
```

Comment:
ideas @Meakk ?

Reply:
see above. I think we should catch it when `set`